lib/model.js

```diff
@@ -3207,19 +3207,28 @@ Model.compile = function compile(name, schema, collectionName, connection, base)
     schema.add(o);
   }
 
-  // generate new class
-  function model(doc, fields, skipId) {
-    if (!(this instanceof model)) {
-      return new model(doc, fields, skipId);
-    }
-    Model.call(this, doc, fields, skipId);
+  var model;
+  if (typeof name === 'function' && name.prototype instanceof Model) {
+    model = name;
+    name = model.name;
```

Comment:
Because `type of name=== 'function'` and `model = name` model is a function. This function.prototype.name is what it's calling. So it would be "myFunction" in the below function `function my function() {}`

With a class this is the name of the class.

Reply:
What if we want our model to have a name different than the class name, i.e if our model is name is 'my.function'?

We have to override it? i.e
```
class myFunction {
     static get name(){
          return 'my.function';
     }
}
```

Reply:
name is a read-only property, so you can't change it
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name

I suggest you look at the link that vkarpov15 commented to use .LoadClass into the schema you want